### ISSUE TITLE
Consider allowing configurable OPTIMISTIC_ABOUT_FILE_LOCKING=x

### ISSUE THREAD

**Issue by aegershman** [splunk/splunk-ansible#349]
I'm running splunk heavy forwarder docker images as apps on Cloud Foundry to use them as HECs to forward logs to our corporate splunk receivers.

[I get the error, `Validating databases (splunkd validatedb) failed with code '1'.` Setting `OPTIMISTIC_ABOUT_FILE_LOCKING=1` in `SPLUNK_HOME/etc/splunk-launch.conf` allows the images to start and function](https://answers.splunk.com/answers/306998/why-am-i-getting-homepathoptsplunkvarlibsplunkaudi.html). (I don't need these HECs to actually write logs to disk or view the UI. I just need them to receiving/transforms/forwarding.)

In order to set this I have to fully overwrite `SPLUNK_HOME/etc/splunk-launch.conf` with just this line. Is there an alternative way to configure `OPTIMISTIC_ABOUT_FILE_LOCKING=x`, or could a configuration flag be enabled to allow setting this? thoughts welcome.

thanks for any time 👍

**Comment by nwang92** [comment]
Hmm I wonder if this is partially related to how CloudFoundry runs containers? The only time I've hit this error is when `/opt/splunk/var` is not backed by an appropriate Docker volume. As in, you'd only get the error if you installed Splunk in a container like:
```
docker run -d -t ubuntu bash
```
but you wouldn't if you installed Splunk in a container like:
```
docker run -d -t -v /opt/splunk/var ubuntu bash
```

In the dockerfiles in docker-splunk, there's a spec in the Dockerfile to always bring up a local volume, even if you don't specify one explicitly: https://github.com/splunk/docker-splunk/blob/develop/splunk/common-files/Dockerfile#L74

I suppose maybe the storage-driver of the docker daemon would also have an impact, but I haven't done any tests to see if that's the case. At the moment, we only support `overlay2` per recommendation from folks at Docker. 

Anyways, not sure if that thinking-out-loud was any helpful :P Is there a way you can share your spec for how these images are deployed on CloudFoundry?

**Comment by nwang92** [comment]
@bb03 and I had a discussion about this, thinking about it more it may make sense to open up ways to add content into `splunk-launch.conf`. In addition to `OPTIMISTIC_ABOUT_FILE_LOCKING`, there are other applications of modifying this file to support things like DFS, Java, etc.

Will leave this issue open, I'll try to work on something for it - I think mostly the challenge here is what is the best way to syntactically define these arbitrary key/value pairs via the default.yml or env vars.

**Comment by aegershman** [comment]
I hear ya, sorry for not getting back to this earlier, these past few weeks have been bonkers; thanks for the consideration, it's not the most important thing on planet earth because I can temporarily manually force the `splunk-launch.conf` to be certain values, but it's not ideal as it overrides anything else set by the ansible automation. Like I said, not the end of planet earth; just looking for an easier way to set up simple Splunk HEC forwarders by leveraging existing components and delivery models via Cloud Foundry, which _will_ run docker images, but is just a bit more stringent about disk writes and filesystem locking, I suppose. Thanks again for the time and consideration 👍

**Comment by nwang92** [comment]
Merged a fix for this - if you use `splunk/splunk:edge` you should be able to do something like:
```
docker run -e SPLUNK_LAUNCH_CONF= OPTIMISTIC_ABOUT_FILE_LOCKING=1 ... splunk/splunk:edge
```

More instructions will be provided through docker-splunk repo, but this should be available in the next release (8.0.2) circa early-to-mid Feb.

**Comment by aegershman** [comment]
awesome, thank you, will try it out 👍
